### ISSUE TITLE
Move SyncerStatus into go-header library

### ISSUE THREAD

**Issue by Manav-Aggarwal** [rollkit/rollkit#1019]
Currently, `HeaderExchangeService` and `BlockExchangeService` use a `SyncerStatus` struct that is created in Rollkit. Ideally, this should be moved inside the go-header library.

**Comment by Wondertan** [comment]
@Manav-Aggarwal, can you please link to SyncerStatus struct? I would like to take a look at it and assess if it's worth the addition to the lib. I wonder if it's something like SyncerState which already exists in the lib.

**Comment by Manav-Aggarwal** [comment]
Of course, [here](https://github.com/rollkit/rollkit/blob/4fd97ba8b8352771f2e66454099785d06fd0c31b/node/header_exchange.go#L28-L31) is the `SyncerStatus` struct in its current form. There's a [PR](https://github.com/rollkit/rollkit/pull/1014) open that moves it [here](https://github.com/rollkit/rollkit/blob/1608fcf9451861f9e6a683e94858037453d40f2b/node/exchange.go#L9).

**Comment by Wondertan** [comment]
So the whole point of this struct is to keep the bool of Syncer's status? Hmm, I don't get why you need it. I found multiple places where the `setStarted  `is used, but not `getStarted` on the commit you shared. We can consider moving it to go-header or, even better make the Syncer report through its state whether it's started or not. For this, we need to open an issue on the go-header repo and briefly describe the use case.

**Comment by Manav-Aggarwal** [comment]
Relevant go-header issue: https://github.com/celestiaorg/go-header/issues/55

The status `started` is checking by `HeaderExchangeService` in its `Stop` method to make sure it's not calling `Stop` on a `SyncerStatus` that hasn't started. Ideally, the Syncer's stop method itself handles this and acts as a no-op in this scenario.

**Comment by Wondertan** [comment]
Thank you @Manav-Aggarwal !

**Comment by tac0turtle** [comment]
issue was opened in go-header closing this
